### ISSUE TITLE
Variables – Default to environment variables

### ISSUE THREAD

**Issue by derekr** [thisandagain/generator#9]
Would make it super fast if I could just hit enter for some of the prompts.

`GENERATOR_AUTHOR=drk` => `__author__`

And something like?

```
> generate -t npm -n test
author (drk):
```

**Comment by thisandagain** [comment]
Agreed. What do you think is a sensible syntax for defaults? I'm not sure if env variables is the right approach because the mapping isn't explicit. Could something like `__author(drk)__` work?

**Comment by thisandagain** [comment]
Unless of course the environment was simply importing an object of defaults. :wink:

**Comment by derekr** [comment]
Oh yeah that'd be cool.

`~/.generator/defaults.json`?

**Comment by derekr** [comment]
Could also pass in as an option per call `generate -t npm -n my-mod -d /some/defaults.json`

**Comment by thisandagain** [comment]
That makes the most sense to me.

Look for `-d /some/defaults.json`
else look for `~/.generator/defaults.json`
else proceed

**Comment by thisandagain** [comment]
Defaults via the CLI should extend the `~/.generator/defaults.json` IMO. Thoughts?

**Comment by derekr** [comment]
Oh yeah that would be really neat.
